endpoints/openrtb2/amp_auction.go

```diff
@@ -574,30 +575,30 @@ func (deps *endpointDeps) overrideWithParams(ampParams amp.Params, req *openrtb2
 	}
 
 	if err := setConsentedProviders(req, ampParams); err != nil {
-		return []error{err}
+		return append(errors, err)
 	}
 
 	policyWriter, policyWriterErr := amp.ReadPolicy(ampParams, deps.cfg.GDPR.Enabled)
 	if policyWriterErr != nil {
-		return []error{policyWriterErr}
+		return append(errors, policyWriterErr)
 	}
 	if err := policyWriter.Write(req); err != nil {
-		return []error{err}
+		return append(errors, err)
 	}
 
 	if ampParams.Timeout != nil {
 		req.TMax = int64(*ampParams.Timeout) - deps.cfg.AMPTimeoutAdjustment
 	}
 
-	if err := setTargeting(req, ampParams.Targeting); err != nil {
-		return []error{err}
+	if warn := setTargeting(req, ampParams.Targeting); warn != nil {
+		errors = append(errors, warn)
 	}
```

Comment:
I'm fine either way on this but I wanted to point out that it might be clearer if `var errors []error` was declared just before this `if` block with the early returns before this just returning `[]error{` since the only time you could potentially collect multiple warnings/errors is from this point forward..

Reply:
Yes, ok, modified.